### PR TITLE
Add specs for SassList and SassMap.

### PR DESCRIPTION
**Pull request by Awjin**
See https://github.com/sass/embedded-host-node/pull/68

[skip sass-embedded]

### CODE REVIEW

File: js-api-spec/value/list.test.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                new SassString('b', {quotes: false}),
+                new SassString('c', {quotes: false}),
+              ],
+              {separator: ',', brackets: true}
+            )
+          )
+        ).toBe(false);
+      });
+
+      it("doesn't equal a value with different contents", () => {
```
Comment (nex3) [inline js-api-spec/value/list.test.ts:92]: Also test equality with a shorter list and a longer list that have the same contents where they overlap.

File: js-api-spec/value/list.test.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        expect(list.hasBrackets).toBe(false);
+      });
+
+      it('supports null separators', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {separator: null}
+        );
+        expect(list.separator).toBe(null);
+      });
```
Comment (nex3) [inline js-api-spec/value/list.test.ts:161]: A list with more than one element shouldn't allow an undecided separator. If that's not in the protocol/spec yet, it should be.

File: js-api-spec/value/list.test.ts [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            string.sassIndexToListIndex(new SassNumber(2))
+          ).toThrow();
+          expect(() =>
+            string.sassIndexToListIndex(new SassNumber(-2))
+          ).toThrow();
+        });
+      });
+    });
+
+    describe('an empty list', () => {
```
Comment (nex3) [inline js-api-spec/value/list.test.ts:244]: Test that this equals another call to `SassList.empty()` and a call to `new SassList()` without any elements.

File: js-api-spec/value/map.test.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      });
+
+      it('rejects invalid indices', () => {
+        expect(() => map.sassIndexToListIndex(new SassNumber(0))).toThrow();
+        expect(() => map.sassIndexToListIndex(new SassNumber(3))).toThrow();
+        expect(() => map.sassIndexToListIndex(new SassNumber(-3))).toThrow();
+      });
+    });
+
+    describe('an empty map', () => {
```
Comment (nex3) [inline js-api-spec/value/map.test.ts:184]: Similar additional equality tests here.

**Review by nex3 — CHANGES_REQUESTED** [review]

File: js-api-spec/value/list.test.ts (hunk 2 above)
Comment (Awjin) [reply js-api-spec/value/list.test.ts]: Good point. I updated the protocol here: https://github.com/sass/embedded-protocol/pull/81. For the spec, would it be easier to update it here https://github.com/sass/sass/pull/3183?

**Review by nex3 — APPROVED** [review]
One more thing: it turns out that initializing values outside of a `beforeEach()` clause breaks the latest version of jest, because it'll try to run them when determining how many tests were skipped. Can you move those into `beforeEach()`es like I've done for the other value tests?

**Comment by Awjin** [comment]
Good catch, done.

**Pull request merged**